crates/bevy_sprite/src/render/mod.rs

```diff
@@ -160,25 +160,31 @@ impl SpritePipelineKey {
     const MSAA_MASK_BITS: u32 = 0b111;
     const MSAA_SHIFT_BITS: u32 = 32 - Self::MSAA_MASK_BITS.count_ones();
 
-    pub fn from_msaa_samples(msaa_samples: u32) -> Self {
+    #[inline]
+    pub const fn from_msaa_samples(msaa_samples: u32) -> Self {
         let msaa_bits =
             (msaa_samples.trailing_zeros() & Self::MSAA_MASK_BITS) << Self::MSAA_SHIFT_BITS;
-        Self::from_bits(msaa_bits).unwrap()
+        // SAFETY: The input is the correct type and the mask will ensure the output range is
+        // capped
+        unsafe { Self::from_bits_unchecked(msaa_bits) }
```

Comment:
For other reviewers here's the safety comment from the bitflags crate

https://docs.rs/bitflags/latest/src/bitflags/lib.rs.html#575-577

```rust
/// The caller of `from_bits_unchecked()` has to ensure that
/// all bits correspond to a defined flag or that extra bits
/// are valid for this bitflags type.
```